examples/rbfe_single.py

```diff
@@ -0,0 +1,160 @@
+
+    parser = argparse.ArgumentParser(
+        description="Relative Hydration Free Energy Consistency Testing",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--num_gpus",
+        type=int,
+        help="number of gpus",
+        required=True
+    )
+
+    parser.add_argument(
+        "--num_complex_windows",
+        type=int,
+        help="number of vacuum lambda windows",
+        required=True
+    )
+
+    parser.add_argument(
+        "--num_solvent_windows",
+        type=int,
+        help="number of solvent lambda windows",
+        required=True
+    )
+
+    parser.add_argument(
+        "--num_equil_steps",
+        type=int,
+        help="number of equilibration lambda windows",
+        required=True
+    )
+
+    parser.add_argument(
+        "--num_prod_steps",
+        type=int,
+        help="number of production lambda windows",
+        required=True
+    )
+
+
+    cmd_args = parser.parse_args()
+
+    multiprocessing.set_start_method('spawn') # CUDA runtime is not forkable
+    pool = multiprocessing.Pool(cmd_args.num_gpus)
+
+    suppl = Chem.SDMolSupplier('tests/data/ligands_40.sdf', removeHs=False)
+    all_mols = [x for x in suppl]
+    mol_a = all_mols[1]
+    mol_b = all_mols[4]
+
+    core = np.array([[ 0,  0],
+       [ 2,  2],
+       [ 1,  1],
+       [ 6,  6],
+       [ 5,  5],
+       [ 4,  4],
+       [ 3,  3],
+       [15, 16],
+       [16, 17],
+       [17, 18],
+       [18, 19],
+       [19, 20],
+       [20, 21],
+       [32, 30],
+       [26, 25],
+       [27, 26],
+       [ 7,  7],
+       [ 8,  8],
+       [ 9,  9],
+       [10, 10],
+       [29, 11],
+       [11, 12],
+       [12, 13],
+       [14, 15],
+       [31, 29],
+       [13, 14],
+       [23, 24],
+       [30, 28],
+       [28, 27],
+       [21, 22]]
+    )
+    ff_handlers = deserialize_handlers(open('ff/params/smirnoff_1_1_0_ccc.py').read())
+    ff = Forcefield(ff_handlers)
+
+    # the water system first.
+    complex_system, complex_coords, _, _, complex_box = builders.build_protein_system('tests/data/hif2a_nowater_min.pdb')
+    complex_box += np.eye(3)*0.1 # BFGS this later
+
+    # the water system first.
+    solvent_system, solvent_coords, solvent_box, _ = builders.build_water_system(4.0)
+    solvent_box += np.eye(3)*0.1 # BFGS this later
+
+
+    for label, host_system, host_coords, host_box, num_host_windows in [
+        ("complex", complex_system, complex_coords, complex_box, cmd_args.num_complex_windows),
+        ("solvent", solvent_system, solvent_coords, solvent_box, cmd_args.num_solvent_windows)]:
+
+        A = int(.35*num_host_windows)
+        B = int(.30*num_host_windows)
+        C = num_host_windows - A - B
+
+        lambda_schedule = np.concatenate([
+            np.linspace(0.0,  0.25, A, endpoint=False),
+            np.linspace(0.25, 0.75, B, endpoint=False),
+            np.linspace(0.75, 1.0,  C, endpoint=True)
+        ])
```

Comment:
Could we include a comment here about the motivation for this lambda schedule? (E.g. a link to a stddev(du/dlambda) profile showing larger du/dlambda in range 0 to 0.25 and a brief statement of the rationale -- want a denser grid of lambda in regions where stddev(du/dlambda) is large. Different lambda schedules appear for different legs in the different examples in this folder, suggesting that some optimization may have been needed here.)

Reply:
I will add a comment that this is a heuristic and is empirical. I'm not sure I'm ready to share plots of this yet until we do more validation.